### PR TITLE
Check existence of docker shared_dir only for local ones

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

File: khiops/extras/docker.py [hunk 1]
```diff
@@ -37,7 +38,9 @@ def __init__(self, url, shared_dir, insecure=False):
         self._url = url
         self._insecure = insecure
         self.shared_dir = shared_dir
-        if self.shared_dir is None or not fs.exists(self.shared_dir):
+        if self.shared_dir is None or (
+            fs.is_local_resource(self.shared_dir) and not fs.exists(self.shared_dir)
+        ):
```
Comment (popescu-v) [inline khiops/extras/docker.py:43]: I would separate the two `if` condition parts and:
- would raise `ValueError` if `self.shared_dir is None`
- would raise `RuntimeError` if `self.shared_dir` is a non-existing local directory

Comment (popescu-v) [inline khiops/extras/docker.py:43]: I would also add a test that a `RuntimeError` (see previous comment) is only raised for a non-existing local `shared_dir`.

**Review by popescu-v — CHANGES_REQUESTED** [review]
Almost LGTM, with two caveats (see comments).

File: khiops/extras/docker.py (hunk 1 above)
Comment (folmos-at-orange) [reply khiops/extras/docker.py:43]: Ok

Comment (folmos-at-orange) [reply khiops/extras/docker.py]: Done

Comment (folmos-at-orange) [reply khiops/extras/docker.py]: Done

Comment (popescu-v) [reply khiops/extras/docker.py]: Ok for these checks.

Comment (popescu-v) [reply khiops/extras/docker.py]: Actually, what I had in mind was an automatic test (in test_remote_access) that a ~`RuntimeError`~ `KhiopsEnvironmentError` is only raised for a non-existing local `shared_dir` (but not for a remote URI).

**Review by popescu-v — CHANGES_REQUESTED** [review]
There is only one comment left to address IMHO.

File: khiops/extras/docker.py (hunk 1 above)
Comment (folmos-at-orange) [reply khiops/extras/docker.py]: That is what i did.

Comment (folmos-at-orange) [reply khiops/extras/docker.py]: Done

File: tests/test_khiops_integrations.py [hunk 2]
```diff
@@ -99,3 +104,13 @@ def test_estimator_multiple_create_and_fit_does_not_raise_exception(self):
         finally:
             if os.path.exists(output_dir):
                 shutil.rmtree(output_dir)
+
+
+class DockerKhiopsEdgeCases(unittest.TestCase):
+    """Test for KhiopsDocker runner edge cases"""
+
+    def test_shared_dir_edge_cases(self):
+        """Test that shared dir exists when it is a local path"""
```
Comment (popescu-v) [inline tests/test_khiops_integrations.py:113]: I would rephrase to something like: "Test that KhiopsEnvironmentError is raised when shared_dir is a local path and does not exist".

File: khiops/extras/docker.py [hunk 3]
```diff
@@ -71,10 +86,12 @@ def _fetch(self, request):
         # HTTPS mode
         response = None
```
Comment (popescu-v) [inline khiops/extras/docker.py:87]: This is not needed anymore IMHO.

File: khiops/extras/docker.py [hunk 4]
```diff
@@ -25,10 +28,30 @@ class KhiopsDockerRunner(KhiopsRunner):
...
 
     Requires a running docker Khiops instance
+
+    Parameters
+    ----------
+
+    url : str
+        URL for the Docker Khiops server.
+    shared_dir : str
+        Path for the shared directory dire. May be an URL.
```
Comment (popescu-v) [inline khiops/extras/docker.py:38]: I would 's/May be an URL/Can be either a local path or an URL/'.

Comment (popescu-v) [inline khiops/extras/docker.py:38]: s/shared directory dire./shared directory./

**Review by popescu-v — CHANGES_REQUESTED** [review]
Almost LGTM, barring 4 remaining comments.

File: tests/test_khiops_integrations.py (hunk 2 above)
Comment (folmos-at-orange) [reply tests/test_khiops_integrations.py:113]: Done

File: khiops/extras/docker.py (hunk 3 above)
Comment (folmos-at-orange) [reply khiops/extras/docker.py:87]: Ok and done

File: khiops/extras/docker.py (hunk 4 above)
Comment (folmos-at-orange) [reply khiops/extras/docker.py:38]: I put "May be an URL/URI" to keep it consistent with the codebase.

**Review by popescu-v — APPROVED** [review]
LGTM
